pkg/build/builder/common.go

```diff
@@ -81,3 +84,45 @@ func updateBuildRevision(c client.BuildInterface, build *api.Build, sourceInfo *
 		glog.Warningf("An error occurred saving build revision: %v", err)
 	}
 }
+
+// runScriptInContainer runs a Bash script in an ephemeral Docker container
+// started off imageID, and returns an error if the script cannot be run or
+// returns a non-zero exit code.
+func runScriptInContainer(client DockerClient, script, imageID string) error {
+	// Read CPU and memory limits from /sys/fs/cgroup.
+	// These limits are used to enforce limits on the container that runs
+	// the post commit hook.
+	r := intFromFileReader{}
+	cpuShares := r.read("/sys/fs/cgroup/cpu,cpuacct/cpu.shares")
+	cpuPeriod := r.read("/sys/fs/cgroup/cpu,cpuacct/cpu.cfs_period_us")
+	cpuQuota := r.read("/sys/fs/cgroup/cpu,cpuacct/cpu.cfs_quota_us")
+	memory := r.read("/sys/fs/cgroup/memory/memory.limit_in_bytes")
+	if r.err != nil {
+		return r.err
+	}
+
+	return dockerRun(client, docker.CreateContainerOptions{
+		// TODO set Name to a reasonable value
+		// (source-to-image/pkg/docker/docker.go doesn't set the Name!)
```

Comment:
We should *always* be setting the container name to something that fits within the pattern of the execution - and we need to put as much detail as possible in the name.  You cannot however use the k8s_ prefix because Kubelet will terminate containers it finds in that space that don't match its name.